lib/screens/home.dart

```diff
@@ -71,92 +79,107 @@ class _HomePageState extends State<HomePage> {
           children: <Widget>[
             Container(
               margin: const EdgeInsets.only(
-                  left: 15, bottom: 20, top: 20, right: 15),
+                left: 15,
+                bottom: 20,
+                top: 20,
+                right: 15,
+              ),
               child: GridView.builder(
-                  scrollDirection: Axis.vertical,
-                  shrinkWrap: true,
-                  physics: const ScrollPhysics(),
-                  itemCount: gfComponents.length,
-                  gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
-                      crossAxisCount: 2,
-                      crossAxisSpacing: 20,
-                      mainAxisSpacing: 20),
-                  itemBuilder: (BuildContext context, int index) =>
-                      buildSquareTile(
-                          gfComponents[index]['title'],
-                          gfComponents[index]['icon'],
-                          gfComponents[index]['route'],
-                          gfComponents[index]['authentication'],
-                          gfComponents[index]['failRoute'])),
+                scrollDirection: Axis.vertical,
+                shrinkWrap: true,
+                physics: const ScrollPhysics(),
+                itemCount: gfComponents.length,
+                gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                    crossAxisCount: 2,
+                    crossAxisSpacing: 20,
+                    mainAxisSpacing: 20),
+                itemBuilder: (BuildContext context, int index) =>
+                    buildSquareTile(
+                  gfComponents[index]['title'],
+                  gfComponents[index]['icon'],
+                  gfComponents[index]['route'],
+                  gfComponents[index]['authentication'],
+                  gfComponents[index]['failRoute'],
+                ),
+              ),
             ),
           ],
         ),
       );
 
   Widget buildSquareTile(String title, IconData icon, Widget route, String auth,
-          Widget failRoute) =>
+          Widget failRoute,
+          {bool enabled = true}) =>
```

Comment:
having an enabler feels kinda dump. can you please switch it with a disabled switch?
you can modify `buildSquareTile(String title, IconData icon, Widget route, String auth, Widget failRoute)` to `buildSquareTile(String title, IconData icon, Widget route, String auth, Widget failRoute, bool disabled )`
pass the `gfComponents[index]['disabled']` for the values and as it's not defined by default the null will be passed and the component will be treated as enabled else it will be disabled.

Reply:
@im-Amitto Yeah. Having 'enabled' feels silly. Don't know why i did that. :sweat:
Anyway, I changed to it to what you suggested.

One thing, on passing null value I got an error saying "boolean expressions can't be null" on the screen. So instead I added a null check in the buildSquareTile method.

```
buildSquareTile(
   gfComponents[index]['title'],
   gfComponents[index]['icon'],
   gfComponents[index]['route'], 
   gfComponents[index]['authentication'],
   gfComponents[index]['failRoute'],
   gfComponents[index]['disabled'] ?? false,
 ),
```

Reply:
noice :)